### PR TITLE
AO3-6032 Expand bookmark controller test

### PR DESCRIPTION
**Pull request by cesy**
## Issue

https://otwarchive.atlassian.net/browse/AO3-6032

## Purpose

Add a few more automated tests, similar to AO3-4923

### CODE REVIEW

File: spec/controllers/bookmarks_controller_spec.rb [hunk 1]
```diff
@@ -219,12 +219,46 @@ def it_redirects_to_user_login
...
         get :show, params: { id: bookmark }
         expect(response).to have_http_status(:success)
         expect(assigns(:bookmark)).to eq(bookmark)
       end
+
+      it "finds a work from a bookmark" do
+        render_views
+        get :show, params: { id: bookmark }
+        expect(response).to have_http_status(:success)
+        expect(response.body).to include(chaptered_work.title)
```
Comment (sarken) [inline spec/controllers/bookmarks_controller_spec.rb:236]: We don't generally render and check the views in specs. Could we check `expect(assigns(:work)).to eq(chaptered_work)` here instead? Or is this aiming for a line in the controller that requires the view?

File: factories/bookmarks.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 FactoryBot.define do
   factory :bookmark do
-    bookmarkable_type { "Work" }
+    bookmarkable_type { "Work" }
```
[the remark this answers is not in the document]
Comment (cesy) [reply factories/bookmarks.rb]: Fixed

File: factories/works.rb [hunk 3]
```diff
@@ -32,6 +32,10 @@
     factory :draft do
       posted { false }
     end
+
+    factory :posted_work do
```
[the remark this answers is not in the document]
Comment (cesy) [reply factories/works.rb]: Fixed

**Comment by cesy** [comment]
Note to self for tomorrow:
Model failure is on the external bookmark: Validation failed: Url could not be reached. If the URL is correct and the site is currently down, please try again later.
     Shared Example Group: "a filterable" called from ./spec/models/concerns/filterable_spec.rb:204
     # ./spec/models/concerns/filterable_spec.rb:203:in `block (2 levels) in <top (required)>'
     # ./spec/models/concerns/filterable_spec.rb:154:in `block (4 levels) in <top (required)>'

Controller failure needs investigating.

File: spec/controllers/bookmarks_controller_spec.rb (hunk 1 above)
Comment (cesy) [reply spec/controllers/bookmarks_controller_spec.rb]: This part is solved now, found a way to check it

File: spec/controllers/bookmarks_controller_spec.rb [hunk 4]
```diff
@@ -252,12 +252,46 @@ def it_redirects_to_user_login
...
+  end
+
+  describe "fetch recent" do
+    # need several bookmarks on the same work
+    let(:work) { create(:work) }
+    let(:bookmark1) { create(:bookmark, bookmarkable_id: work.id, bookmarker_notes: "Extra 1") }
+    let!(:bookmark2) { create(:bookmark, bookmarkable_id: work.id, bookmarker_notes: "Different") }
+    let!(:bookmark3) { create(:bookmark, bookmarkable_id: work.id, bookmarker_notes: "I think this is great") }
+
+    # wouldn't normally need to render views in a controller test, but it's the only way to get the js response
```
Comment (sarken) [inline spec/controllers/bookmarks_controller_spec.rb:281]: We could actually leave it out -- the JavaScript functionality is already covered by [a feature test](https://github.com/otwcode/otwarchive/blob/b63fce67a3975ec0614bb535c75d76c8dafbda71/features/bookmarks/bookmark_create.feature#L389-L411). It's just the [non-JavaScript lines this adds new coverage for](https://app.codecov.io/gh/otwcode/otwarchive/compare/3890/changes#D1L305).

**Comment by Bilka2** [comment]
#4808 ([AO3-6725](https://otwarchive.atlassian.net/browse/AO3-6725)) removed `fetch_recent`, so that test is no longer needed.

I think this PR and the related Jira issue can be closed. The test coverage of bookmarks_controller is 99.51% on master, the only uncovered line is a [very generic error](
https://app.codecov.io/gh/otwcode/otwarchive/blob/master/app%2Fcontrollers%2Fbookmarks_controller.rb#L219).

**Pull request closed without merging**